### PR TITLE
Versioned releases

### PR DESCRIPTION
**Pull request by msecret**
Things done:
    - Added manifest for master, staging and deprecated
    - Exit out of build if no tag (not 100 sure if this works)
    - Added script to get npm version number.
    - Put npm version number in build information
    - Updated some old references in readme and contributing, but don't want
      to focus on documentation updates right now.
    - Named the app in package.json with starting 1.0.0 version

    Still to do:
    - Create tag for version 1.0.0, maybe do this after this branch is
    merged.

### CODE REVIEW

**Comment by msecret** [comment]
@jcscottiii I also cherry-picked in ArthurHlt's work

**Comment by jcscottiii** [comment]
@msecret  for the Arthur's commit, are we using this PR or #407?
I pushed a commit to fix part of the failures. 
The other failures will be fixed once #414 is merged

**Comment by jcscottiii** [comment]
ahh you need to add `tags: true`
https://docs.travis-ci.com/user/deployment

File: deploy/travis_deploy.sh [hunk 1]
```diff
@@ -12,12 +12,21 @@ set -e
 CF_PATH="."
 if [ "$TRAVIS_BRANCH" == "master" ]
 then
-	CF_MANIFEST="manifests/manifest-master.yml"
-	CF_SPACE="deck-prod"
+  CF_MANIFEST="manifests/manifest-prod.yml"
+  CF_SPACE="deck-prod"
+  # Skip out of build if no tag
+  if ! [[ "$TRAVIS_TAG" =~ ^v[0-9]+\.[0-9]+\.[0-9]+(-[A-Za-z0-9-]+)? ]]; then
```
Comment (jcscottiii) [inline deploy/travis_deploy.sh]: will the branch be set if a tag is committed ?

Comment (jcscottiii) [inline deploy/travis_deploy.sh]: also, the file uses tabs (:( i know)

Comment (jcscottiii) [inline deploy/travis_deploy.sh]: whats the format of the tag? is this regex looking for semver? if so, put a comment clarifying.

Comment (jcscottiii) [inline deploy/travis_deploy.sh]: you might want to move the Travis Tag check out and put it first in this if chain.

**Pull request merged**